feedback-suggestion/src/app/endpoints/feedback_suggestions.py

```diff
@@ -46,13 +46,25 @@ def get_feedback_suggestions(request: FeedbackSuggestionsRequest):
     suggested_feedbacks = []
     for filepath, methods in function_blocks.items():
         for feedback in db_feedbacks:
+            if feedback.participation_id == request.participation_id:
+                continue
             if feedback.src_file == filepath:
                 for method in methods:
                     # F1 is the similarity score, F3 is similar to F1 but with a higher weight for recall than precision
                     precision, recall, F1, F3 = score(cands=[method.get_source_code()], refs=[feedback.code],
                                                       lang='java')
-                    similarity_score = torch.mean(F1)  # TODO: is this correct?
+                    similarity_score = float(torch.mean(F1))  # TODO: is this correct?
                     if similarity_score >= SIMILARITY_SCORE_THRESHOLD:
-                        suggested_feedbacks.append(feedback)
+                        print(f"Found similar code with similarity score {similarity_score}: {feedback}")
+                        original_code = feedback.code
+                        feedback_to_give = dict(feedback)
+                        feedback_to_give["code"] = method.get_source_code()
+                        feedback_to_give["from_line"] = method.get_start_line()
+                        feedback_to_give["to_line"] = method.get_stop_line()
+                        suggested_feedbacks.append({
+                            "feedback": feedback_to_give,
+                            "originally_on_code": original_code,
+                            "similarity_score": similarity_score
+                        })
```

Comment:
Why send the code twice? Technically we don't need the code at all in Themis. The same goes for the similarity score. I would just send what really is needed, and that is the feedback.

Reply:
One is the code from the submission in question, the other is the code from the feedback in the DB. They are the two different pieces of code compared by codeBERT. This is useful for debugging, even though our app does not need it. I will add a boolean parameter which decides whether to include the code and set it to `False` by default.